v2/internal/reconcilers/arm/arm_client_cache_test.go

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package arm
+
+import (
+	"context"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/go-autorest/autorest/to"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	. "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	resources "github.com/Azure/azure-service-operator/v2/api/resources/v1beta20200601"
+	"github.com/Azure/azure-service-operator/v2/internal/config"
+	"github.com/Azure/azure-service-operator/v2/internal/genericarmclient"
+	"github.com/Azure/azure-service-operator/v2/internal/metrics"
+	"github.com/Azure/azure-service-operator/v2/internal/resolver"
+	"github.com/Azure/azure-service-operator/v2/internal/util/kubeclient"
+)
+
+const (
+	fakeID = "00000000-0000-0000-0000-000000000000"
+)
+
+func NewFakeKubeClient(s *runtime.Scheme) kubeclient.Client {
+	fakeClient := fake.NewClientBuilder().WithScheme(s).Build()
+	return kubeclient.NewClient(fakeClient)
+}
+
+func NewTestArmClientCache(client kubeclient.Client) (*armClientCache, error) {
+	cfg, err := config.ReadFromEnvironment()
+	if err != nil {
+		return nil, err
+	}
+
+	creds, err := azidentity.NewDefaultAzureCredential(nil)
+	if err != nil {
+		return nil, err
+	}
+
+	globalARMClient, err := genericarmclient.NewGenericClient(cfg.Cloud(), creds, cfg.SubscriptionID, metrics.NewARMClientMetrics())
+	if err != nil {
+		return nil, err
+	}
+
+	return NewARMClientCache(globalARMClient, cfg.PodNamespace, client, cfg.Cloud(), nil), nil
+}
+
+type testResources struct {
+	armClientCache *armClientCache
+	kubeClient     kubeclient.Client
+}
+
+func createTestScheme() *runtime.Scheme {
+	s := runtime.NewScheme()
+
+	_ = v1.AddToScheme(s)
+	_ = resources.AddToScheme(s)
+
+	return s
+}
+
+func testSetup() (*testResources, error) {
+	s := createTestScheme()
+
+	client := NewFakeKubeClient(s)
+	cache, err := NewTestArmClientCache(client)
+	if err != nil {
+		return nil, err
+	}
+
+	return &testResources{
+		armClientCache: cache,
+		kubeClient:     client,
+	}, nil
+}
+
+func Test_ARMClientCache_ReturnsNamespacedSecretClient(t *testing.T) {
```

Comment:
minor: Would calling this `Test_ARMClientCache_ReturnsNamespaceScopedClient` be slightly clearer? IIRC the client it's returning isn't a `SecretClient`, which is sorta what this test makes me think at first glance.